dependencies.yaml

```diff
@@ -57,16 +57,16 @@ dependencies:
           - cython>=0.29,<0.30
           - dlpack>=0.5,<0.6.0a0
           - ninja
-          - &pyarrow_ver pyarrow=9.0.0
+          - pyarrow=10
           - rmm=23.02.*
           - scikit-build>=0.13.1
       - output_types: conda
         packages:
-          - arrow-cpp=9
+          - libarrow=10
           - c-compiler
           - cxx-compiler
           - librdkafka=1.7.0
-          - protobuf>=3.20.1,<3.21.0a0
+          - protobuf=4.21
```

Comment:
Porting my protobuf question for @ajschmidt8 @bdice: do we expect any issues with this hard pinning?